File: basearchcomponents/src/main/java/com/tanveershafeeprottoy/basearchcomponents/viewmodels/BaseAndroidViewModel.java
package com.tanveershafeeprottoy.basearchcomponents.viewmodels;

import android.app.Application;

import com.tanveershafeeprottoy.basearchcomponents.livedata.SingleLiveEvent;

import androidx.lifecycle.AndroidViewModel;

/**
 * @author Tanveer Shafee Prottoy
 */
public abstract class BaseAndroidViewModel extends AndroidViewModel {
    protected SingleLiveEvent<Throwable> throwableSingleLiveEvent = new SingleLiveEvent<>();

    public BaseAndroidViewModel(Application application) {
        super(application);
    }

    protected void throwError(Throwable throwable) {
        throwableSingleLiveEvent.setValue(throwable);
    }
}

File: basearchcomponents/src/main/java/com/tanveershafeeprottoy/basearchcomponents/viewmodels/BaseAndroidViewModel.kt
package com.tanveershafeeprottoy.basearchcomponents.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.tanveershafeeprottoy.basearchcomponents.interfaces.BaseViewModel
import com.tanveershafeeprottoy.basearchcomponents.livedata.SingleLiveEvent

abstract class BaseAndroidViewModel(application: Application) : AndroidViewModel(application), BaseViewModel {
    override val throwableSingleLiveEvent: SingleLiveEvent<Throwable?> = SingleLiveEvent()

    override fun throwError(throwable: Throwable?) {
        throwableSingleLiveEvent.value = throwable
    }
}